File: 001-100/002/evenFibonacciNumbers.cpp
#include <iostream>
#include <vector>
using namespace std;

// Generate the Fibonacci Sequence
std::vector<int> fibonacciSequence(int n){
  std::vector<int> sequence; int next;
  sequence.push_back(1); sequence.push_back(2);
  for(int i = 2; i < n; i++){
    next = sequence[i - 1] + sequence[i - 2];
    if(next > n){ break; }
    else{ sequence.push_back(next); }
  }
  return sequence;
}
// Calculate sum of even-valued terms
int evenFibonacciNumbers(int n){
  int sum = 0;
  std::vector<int> sequence = fibonacciSequence(n);
  for(int i = 0; i < sequence.size(); i++){
    if(sequence[i] % 2 == 0){ sum += sequence[i]; }
  }
  return sum;
}

int main(){
  // Get user input
  int n;
  cout << "Enter a max fibonacci value n: ";
  cin >> n;
  int fibSum = evenFibonacciNumbers(n);
  // Print results
  cout << "Sum of even-valued fibonacci terms below " << n << endl;
  cout << fibSum << endl;
  return 0;
}

File: 001-100/002/evenFibonacciNumbers.php
<?php
  // Generate the Fibonacci Sequence
  function maxFibonacciValue($n){
    $fib = array(1, 2);
    for ($i=2; $i < $n; $i++) {
      $next = $fib[$i - 1] + $fib[$i - 2];
      if($next > $n){ break; }
      else{ array_push($fib, $next); }
    }
    return $fib;
  }
  // Calculate sum of even-valued terms
  function evenFibonacciNumbers($n){
    $sum = 0;
    $seq[] = maxFibonacciValue($n);
    foreach($seq[0] as $value){
      if($value % 2 == 0){ $sum += $value; }
    }
    return $sum;
  }
  // Get user input
  $prompt = "Enter a max fibonacci value n: ";
  $n = readline($prompt);
  $fibSum = evenFibonacciNumbers($n);
  // Print results
  echo "\nSum of even-valued fibonacci terms below $n";
  echo "\n$fibSum\n";
?>
